src/openms/source/ANALYSIS/QUANTITATION/PeptideAndProteinQuant.cpp

```diff
@@ -94,12 +93,6 @@ namespace OpenMS
       PeptideData& data = pep_quant_[hit.getSequence()];
       data.psm_count++;
 
-      // TODO: why is this needed
-      for (Size i = 1; i <= n_fractions; ++i)
-      {
-        data.abundances[i][hit.getCharge()]; // insert empty element for charge
```

Comment:
Does this maybe influence what columns are generated? If charges are treated separately

Reply:
I have not tested different charges. Is there a test?
But generally we use [] to access things everywhere in this class. So it will be initialized with zero as needed later on.
I don't think there is a need to allocate everything here.

Reply:
yeah but maybe somewhere it is used to count the number of different charge states. Will look for .size()

Reply:
https://github.com/OpenMS/OpenMS/blob/1ed9ca0e9a047331368a15c2f381c2067a1ebbf8/src/openms/source/ANALYSIS/QUANTITATION/PeptideAndProteinQuant.cpp#L163

Reply:
I think it is used and removing it will cause an error

Reply:
maybe not ;)